From 851d9387e0489ab939b3b4aa7599bae8e5a42144 Mon Sep 17 00:00:00 2001
From: Kevin Hao <kexin.hao@windriver.com>
Date: Thu, 22 Sep 2011 12:58:59 +0800
Subject: [PATCH] wrhv: don't migrate the NO_BALANCING irqs when reboot

Some device drivers such as QMAN/BMAN are using IRQF_NO_BALANCING
flag to disable the migration of these irqs to other cores because
these irqs are core specific, and their corresponding irq handlers
also don't support reentry. We already added a check to avoid these
migrations when a cpu down. This patch also adds such a check for
reboot.

Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 kernel/vbi/wrhv.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/kernel/vbi/wrhv.c b/kernel/vbi/wrhv.c
index 880ed63..75fa098 100644
--- a/kernel/vbi/wrhv.c
+++ b/kernel/vbi/wrhv.c
@@ -714,7 +714,7 @@ static int wrhv_reset_all_irqs(struct notifier_block *self,
 
 		if (!desc)
 			continue;
-		if (desc->status & IRQ_PER_CPU)
+		if (desc->status & (IRQ_PER_CPU | IRQ_NO_BALANCING))
 			continue;
 
 		raw_spin_lock(&desc->lock);
-- 
1.7.0.4

